File: src/main/java/cn/woyeshi/server/mapper/UserMapper.java
package cn.woyeshi.server.mapper;

import cn.woyeshi.server.domain.User;
import cn.woyeshi.server.domain.UserExample;

import java.util.List;

import org.apache.ibatis.annotations.Mapper;
import org.apache.ibatis.annotations.Param;
import org.springframework.stereotype.Service;

@Service("userMapper")
@Mapper
public interface UserMapper {
    long countByExample(UserExample example);

    int deleteByExample(UserExample example);

    int deleteByPrimaryKey(String userId);

    int insert(User record);

    int insertSelective(User record);

    List<User> selectByExample(UserExample example);

    User selectByPrimaryKey(String userId);

    int updateByExampleSelective(@Param("record") User record, @Param("example") UserExample example);

    int updateByExample(@Param("record") User record, @Param("example") UserExample example);

    int updateByPrimaryKeySelective(User record);

    int updateByPrimaryKey(User record);
}

File: src/main/kotlin/cn/woyeshi/server/mapper/UserMapper.kt
package cn.woyeshi.server.mapper

import cn.woyeshi.server.domain.User
import cn.woyeshi.server.domain.UserExample

import org.apache.ibatis.annotations.*
import org.springframework.stereotype.Service

@Service(value = "userMapper")
@Mapper
interface UserMapper {
    fun countByExample(example: UserExample): Long

    fun deleteByExample(example: UserExample): Int

    fun deleteByPrimaryKey(userId: String): Int

    fun insert(record: User): Int

    fun insertSelective(record: User): Int

    fun selectByExample(example: UserExample?): List<User>

    fun selectByPrimaryKey(userId: String): User

    fun updateByExampleSelective(@Param("record") record: User, @Param("example") example: UserExample): Int

    fun updateByExample(@Param("record") record: User, @Param("example") example: UserExample): Int

    fun updateByPrimaryKeySelective(record: User): Int

    fun updateByPrimaryKey(record: User): Int

    @Select("Select * from t_user where user_name = '\${userName}' and password = '\${password}'")
    @Results(Result(property = "userName", column = "user_name"), Result(property = "userId", column = "user_id"))
    fun selectByUserNameAndPwd(@Param("userName") userName: String, @Param("password") password: String): User
}